### PR TITLE
Payout → Contributions refactor: Participation as first-class lifecycle

### PR DESCRIPTION
**Pull request by mellelieuwes**
Flux: https://3.basecamp.com/5734045/buckets/35926565/todos/10164216570

## Summary

Replaces the PayoutModal + task-level accept/reject flow with a **Participation-driven** model. `Assignment.Participation` is now the entity that owns the completed/accepted/rejected lifecycle; reward transitions commit inside the same Multi as the participation update; owner UX consolidates on a new Contributions tab + PendingContributions next-best-action.

## What changed

**Domain shift**
- `Assignment.Instance` → `Assignment.Participation` with `completed_at` / `accepted_at` / `rejected_at` / `rejected_message` fields.
- `Assignment.Public.{complete,accept,reject}_participation/1..2` are the funnel; each fires a `{:assignment_participation, :*}` signal.
- Business rule: **accept after reject is blocked** (`{:error, :participation_already_rejected}`) — rejection reopens the assignment spot; dispute resolution happens outside the system.

**Fund API**
- `Fund.Public.mark_pending_approval / approve_reward / reject_reward` now take `(multi, %Fund.RewardModel{})` — callers compose atomically.
- `approve_reward` on a `:rejected` reward is currency-aware: only `:legal` enforces `fund.available >= amount`; virtual currencies skip the guard (mirrors `guard_fund_balance` on reserve).
- Deleted: `Assignment.Public.payout_participant/2`, task-level accept/reject helpers, `Crew.RejectView/RejectModel/RejectCategories`.

**UI**
- Contributions tab (split into Pending / Confirmed / Declined sub-views) replaces PayoutModal.
- `PendingContributions` next-best-action surfaced on the owner's dashboard when contributions await review.
- Free-text decline reason (categorized reject removed).

**Layering / hygiene**
- `Assignment.Switch` intercept clauses for the three new signals do only post-commit housekeeping (NA create/clear + view refresh); reward flips happen inside the Multi.
- `reward_idempotence_key` moved from `Assignment.Public` (defp) to `Assignment.Private` (public), imported by Public.
- Test helpers: `Systems.Assignment.TestHelper.create_reward/3` + `Systems.Fund.TestHelper.{mark_pending_approval,approve_reward,reject_reward}/1`.

**Tests added**
- Assignment.Switch: 8 tests for `:completed / :accepted / :rejected` intercepts (NA create/drain, fund/owner guards, empty-owners edge case).
- Public: state-transition matrix (accept-after-reject block, reject-after-accept block), bulk `:confirm_all` + NA drain across N participations.
- Wallaby: `approve_participation_test.exs` exercises the real signal chain end-to-end.
- Feature suite alignment: stale testids updated, signin races fixed across 7 tests.

## Test plan

- [x] `mix test.unit` — green
- [x] `mix test.feature` — 25 features, 0 failures
- [x] `mix compile --warnings-as-errors` — clean
- [x] `mix credo` — clean
- [x] `mix dialyzer` — clean
- [ ] Manual smoke test in staging once merged: participant completes task → owner sees NA on dashboard → clicks NA → confirms contribution → NA clears; also participant → declines flow.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: core/systems/crew/_public.ex [hunk 1]
```diff
@@ -125,13 +137,13 @@ defmodule Systems.Crew.Public do
...
+    |> order_by([task: t], desc: t.updated_at)
     |> Repo.all()
   end
 
   def accepted_tasks(crew) do
     task_query(crew, [:accepted])
-    |> order_by([task: t], desc: t.accepted_at)
+    |> order_by([task: t], desc: t.updated_at)
     |> Repo.all()
   end
```
Comment (JoeriDijkstra) [inline core/systems/crew/_public.ex]: Are these still relevant if we remove the whole rejecting and accepting of tasks? I don't see any callers when ripgrepping the code

File: core/systems/fund/_public.ex [hunk 2]
```diff
@@ -375,56 +365,60 @@ defmodule Systems.Fund.Public do
...
+    end)
+    |> approve_payment_step(reward)
+    |> cas_approve_step(reward, [:rejected], rejected_at: nil)
   end
 
-  defp do_approve_reward(%Fund.RewardModel{} = reward) do
-    Multi.new()
+  def approve_reward(%Multi{} = multi, %Fund.RewardModel{status: status} = reward)
+      when status in [:reserved, :pending_approval] do
+    multi
```
Comment (JoeriDijkstra) [inline core/systems/fund/_public.ex:407]: Possible status is one of `@statuses [:reserved, :pending_approval, :approved, :pending_payout, :rejected, :paid]` - I don't see :pending_payout covered so then it'll crash when called

**Review by JoeriDijkstra — APPROVED** [review]
Small things that shouldn't block the branch from being merged, but would be nice to be picked up in a follow up

File: core/systems/crew/_public.ex (hunk 1 above)
Comment (mellelieuwes) [reply core/systems/crew/_public.ex]: Nice find, will drop this dead code

File: core/systems/fund/_public.ex (hunk 2 above)
Comment (mellelieuwes) [reply core/systems/fund/_public.ex:407]: Will fix this
